src/validation/rules/optional/ProhibitIntrospectionQueriesRule.js

```diff
@@ -0,0 +1,41 @@
+// @flow strict
+
+import { GraphQLError } from '../../../error/GraphQLError';
+
+import { type FieldNode } from '../../../language/ast';
+import { type ASTVisitor } from '../../../language/visitor';
+
+import { type ValidationContext } from '../../ValidationContext';
+
+import { getNamedType } from '../../../type/definition';
+import { isIntrospectionType } from '../../../type/introspection';
```

Comment:
You should order imports in hierarchical so `../../../type` should be before `../../ValidationContext`.
So first ordering criteria.

We put type import value imports in the same import statements so it doesn't make sense to order by them by type/non-type criteria since latter you would mix them in the same statement anyway.

Reply:
Totally missed that's the code style being enforced. Would it make sense to update the [import/order](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/order.md) ESLint rule to help enforce that?

Reply:
If you can do that would be great at the same time I don't think we can enforce everything with their current config.
For example, I order imports in order (it matches order of chapters in GraphQL spec):
* polyfills
* \__testUtils__
* jsutils
* error
* **\<same order as chapters in graphql spec>**
* language
* type
* validation
* execution
* subscription
* **\<end of the same order>**
* utilities
* `src/graphql.js`
* files in upper dir
* files in the same dir
* \__fixtures__

If we can enforce something like this with `import/order` I'm totally for it if not I think we need to write our own custom rule.
I'm also ok to change the order if it would be easier to enforce it but for folders that present spec chapters like to keep them in the same order.

@danielrearden Can you please open an issue to track this?

Reply:
Also, it definitely not compete with TS migration but actually help with reviewing such PRs.